src/Microsoft.ML.ImageAnalytics/ImageResizer.cs

```diff
@@ -298,11 +298,6 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                         getSrc(ref src);
                         if (src == null || src.Height <= 0 || src.Width <= 0)
                             return;
-                        if (src.Height == info.ImageHeight && src.Width == info.ImageWidth)
-                        {
```

Comment:
This seems to be a an optimization that is ignoring pixel format. Can you retain this optimization if you also compare pixel formats of source and destination?

Reply:
The source already has the Pixel Format, so wouldn't the destination have it by default? Or am I misunderstanding?

Reply:
Something like this:
`if (src.Height == info.ImageHeight && src.Width == info.ImageWidth && src.PixelFormat == info.PixelFormat)`



---
In reply to: [517678263](https://github.com/dotnet/machinelearning/pull/5474#discussion_r517678263) [](ancestors = 517678263)

Reply:
Basically we need to check the `PixelFormat` of the source to see if it is supported. When requested size is the same as source size, we are returning the source bitmap directly. But if the image contains a pixel format not supported by `Graphics.DrawImage` (as described in lines 384-388) then there is an exception. 
So we need to check the PixelFormat (see lines 389-392) and return the source only if the PixelFormat is supported. (You can consider refactoring that if statement into a separate function `IsPixelFormatSupported` if you wish)

---
In reply to: [517686588](https://github.com/dotnet/machinelearning/pull/5474#discussion_r517686588) [](ancestors = 517686588,517678263)

Reply:
Oh I see what you mean, but the format the user provided is `Format32bppArgb`,  which is supported, and that is only needed by `g.DrawImage`, which is only needed if you want to create a new bitmap. I think is necessary to create another bitmap because the `src` Bitmap is modified and reused again, which causes the different results and exception.

Reply:
I updated the description. The exception is caused because the source image gets disposed within other transforms that try loading the image again.

Reply:
I think the simpler way to resolve this issue is to create another bitmap, but if you have other suggestions, let me know.

Reply:
This is not the right fix. The creator of the object should manage its lifetime. When the dimensions are different, ImageResizer creates a new Bitmap and disposes it in the next call (and not in an obvious way) by disposing off dst which just happens to be a side effect of the next transformer reusing its variable.  
When the dimensions are the same, it is returning the same src bitmap it got from the previous transformer. In this case it does not own it and since it did not create it, it should not dispose it off in the next call when that parameter comes in as the dst parameter. It should dispose the dst only if it is not the same as the previous src bitmap (implying that this was a bitmap created by the ImageResizer.)